File: lib/src/main/java/com/sd/lib/http/callback/StringRequestCallback.java
package com.sd.lib.http.callback;

public abstract class StringRequestCallback extends RequestCallback
{
    private String mResult;

    @Override
    public void onSuccessBackground() throws Exception
    {
        super.onSuccessBackground();
        mResult = getResponse().getAsString();
    }

    /**
     * 返回请求的字符串
     *
     * @return
     */
    public final String getResult()
    {
        return mResult;
    }
}

File: lib/src/main/java/com/sd/lib/http/callback/StringRequestCallback.kt
package com.sd.lib.http.callback

import com.sd.lib.http.IResponse

abstract class StringRequestCallback : RequestCallback() {
    /** 返回请求的字符串内容，onSuccessXXXX方法中不为null */
    var result: String? = null
        private set

    @Throws(Exception::class)
    override fun onSuccessBackground(response: IResponse) {
        super.onSuccessBackground(response)
        result = response.readString()
    }
}